### ISSUE TITLE
expo build:web results in a blank web page

### ISSUE THREAD

**Issue by aejaex** [expo/expo#5625]
### Environment

  Expo CLI 3.0.10 environment info:
    System:
      OS: Windows 10
    Binaries:
      Yarn: 1.17.3 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
      npm: 6.9.0 - C:\Program Files\nodejs\npm.CMD

I am expo SDK 34 and I have tested on Google Chrome and Internet Explorer.

### Steps to Reproduce

I simply created a new expo project using "expo init newProject", and then I ran expo build:web.

A folder "web-build" was generated, however when I run index.html on my browser I get a blank page.

### Expected Behavior

The default "Open up App.js to start working on your app" text should appear on the browser. 

### Actual Behavior

I get a blank page on the browser after running index.html from the web-build folder.

Also the following error is displayed on the console:

TypeError: Failed to register a ServiceWorker: The URL protocol of the current origin ('null') is not supported.

However when I run expo start --web it works fine.

### Reproducible Demo

Simply an expo init newProject and then expo build:web should suffice.

**Comment by cruzach** [comment]
@willcrisis could you share your package.json? I recently tested deploying through GH pages and didn't see that issue

**Comment by willcrisis** [comment]
@cruzach 

```
{
  "main": "node_modules/expo/AppEntry.js",
  "scripts": {
    "predeploy": "expo build:web",
    "deploy": "gh-pages -d web-build",
    "start": "expo start",
    "android": "expo start --android",
    "ios": "expo start --ios",
    "web": "expo start --web",
    "eject": "expo eject",
    "test": "jest --watchAll"
  },
  "jest": {
    "preset": "jest-expo"
  },
  "dependencies": {
    "@expo/samples": "~3.0.3",
    "@expo/vector-icons": "^10.0.3",
    "@react-navigation/web": "^1.0.0-alpha.9",
    "expo": "^34.0.1",
    "expo-asset": "^6.0.0",
    "expo-constants": "6.0.0",
    "expo-font": "~6.0.0",
    "expo-web-browser": "6.0.0",
    "react": "16.8.3",
    "react-dom": "^16.8.6",
    "react-native": "https://github.com/expo/react-native/archive/sdk-34.0.0.tar.gz",
    "react-native-gesture-handler": "~1.3.0",
    "react-native-web": "^0.11.4",
    "react-navigation": "^3.11.0"
  },
  "devDependencies": {
    "babel-preset-expo": "^6.0.0",
    "gh-pages": "^2.1.1",
    "jest-expo": "^34.0.0"
  },
  "private": true,
  "homepage": "https://willcrisis.github.io/expo-test"
}
```

The only changes I made were adding the `homepage` value, and the `predeploy` and `deploy` scripts.

**Comment by EvanBacon** [comment]
https://github.com/expo/expo-cli/pull/938

**Comment by willcrisis** [comment]
@cruzach @EvanBacon Github Pages deployment looks fixed from expo-cli version 3.1 on.

**Comment by elie222** [comment]
> Have you tried deploying it through [one of these methods](https://docs.expo.io/versions/latest/distribution/publishing-websites/)?

Facing a similar issue. We don't want to deploy using one of those methods.

Tried running it with `serve` locally, but didn't have any luck.
